app/src/main/java/com/ubergeek42/WeechatAndroid/fragments/BufferFragment.kt

```diff
@@ -64,15 +64,8 @@ import com.ubergeek42.WeechatAndroid.upload.i
 import com.ubergeek42.WeechatAndroid.upload.insertAddingSpacesAsNeeded
 import com.ubergeek42.WeechatAndroid.upload.suppress
 import com.ubergeek42.WeechatAndroid.upload.validateUploadConfig
+import com.ubergeek42.WeechatAndroid.utils.*
```

Comment:
>Index is saved per-buffer, but we are using a global array of sent messages. Sending messages in another buffer will make index in current one invalid.

I don't think this is a real issue since we don't have per-buffer history. What do you suggest instead?

>When switching between messages the cursor flickers to start of the string and to end, this can probably be avoided?

Resolved by removing `post`.

>when you navigate up AND edit a previous message AND navigate elsewhere, that last input is NOT preserved

That's correct. I don't think I want to fix that, that would basically mean introducing some kind of forking history. At which point do you forget these edits of past sent messages? Do you add a new version in-between that's preserved in `sentMessages` somehow?

>Moreover, if you send it, the NEXT input that you type in is NOT preserved when navigating away. This is not an edge case and should be addressed.

Ah nice catch. That's because I don't reset the index when sending.